docs/user-guide/FAQ.md

```diff
@@ -128,3 +128,10 @@ By default, step commands are evaluated with the Bourne shell (`/bin/sh`). You c
 
 ## How do I speed up time to upload artifacts?
 You can set the environment variable [`SD_ZIP_ARTIFACTS`](./environment-variables#user-configurable) to `true` which will zip artifacts before uploading, provided your cluster admin has set it up properly.
+
+## How do I disable shallow cloning?
+You can set the environment variable [`GIT_SHALLOW_CLONE`](./environment-variables#user-configurable) to `false` in order to disable shallow cloning.
+
+By default Screwdriver will shallow clone source repositories to a depth of 50. Screwdriver will also enable the `--no-single-branch` flag by default.
```

Comment:
should this be behind a config ?

Reply:
I'm not seeing any noticeable difference whether this flag is enabled or disabled when testing a large repo (~160k commits).

Reply:
On a separate note, the branching logic for `getCheckoutCommand` is slowly getting more complex. At some point we may want to consider refactoring the way we approach bookends.